Mimus/Source/Verification/MismatchMessageCreator.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  Copyright © 2019 AirHelp. All rights reserved.
+//
+
+import Foundation
+
+class MismatchMessageCreator {
+    
+    func message(for mismatchedResults: [MatchResult]) -> String {
+        guard !mismatchedResults.isEmpty else {
+            return ""
+        }
+        let mismatchedCallsCountInfo = "Found \(mismatchedResults.count) call(s) with expected identifier, but not matching arguments:"
```

Comment:
This gets a bit too wordy IMO in final message:

```
failed - Call with identifier Identifier was recorded 1 times, but arguments didn't match. Found 1 call(s) with expected identifier, but not matching arguments:
Mismatched call #1:
Mismatch in argument #2 - expected <(["type": "logina"])>, but was <(["type": "login"])>.
```

We're now basically repeating the same message twice. Verifying for a single a call will be the most commonly used scenario so I'd make the message as short as possible. Ideally this should state:

```
failed - Call with identifier Identifier was recorded 1 times, but arguments are different.

Mismatch in argument #2 - expected <(["type": "logina"])>, but was <(["type": "login"])>.
```

And for longer messages we currently have:

```
failed - Call with identifier was recorded 1 times, but expected 2. Found 1 call(s) with expected identifier, but not matching arguments:
Mismatched call #1:
Mismatch in argument #0 - expected <(Fixture Email)>, but was <(Fixture Emaila)>.
```

I'd alter this to (more concise):

```
failed - Call with identifier was recorded 1 times, but expected 2. 1 additional call(s) matched identifier, but not arguments:
Mismatched call #1:
Mismatch in argument #0 - expected <(Fixture Email)>, but was <(Fixture Emaila)>.
```

Reply:
I've made the changes to the messages that you mentioned. I'm not sure if I got it exactly the way that you noted, but I think that it should cover mentioned issues.
1. If it turns out that there is only one recorded call, then we do not add the `Mismatched call #1:` messages above the listed arguments as there is no point in this.  

2. In case of longer messages, we will have the `1 additional call(s) matched identifier, but not arguments:` message.

3. I still have a small issue with assigning numbers to mismatched arguments as I feel that counting from 0 in this case might not be that intuitive. WDYT?

Reply:
It's perfect now. Merging! 🚀 

One minor nitpick I'll add manually is a bit of newline spacing to get the message a bit easier to parse using eyes. 

As per point 3. I think it's better to have 0-indexed indexing as it's a thing that programmers are more accustomed to.